src/base-type.ts

```diff
@@ -309,7 +305,7 @@ export abstract class BaseTypeImpl<ResultType, TypeConfig = unknown> implements
      *
      * @param name - the new name to use in error messages
      */
-    withBrand<BrandName extends string>(name: BrandName): Type<Branded<ResultType, BrandName>, TypeConfig> {
+    withBrand<const BrandName extends string>(name: BrandName): Type<Branded<ResultType, BrandName>, TypeConfig> {
```

Comment:
Somehow, this prevents TypeScript from analyzing too deep. Before this change, it would complain about `x.withBrand().literal().autocast` not being compatible with `x` or something like that. Now it doesn't. So apparently, the compiler only considers this type parameter on a case-by-case basis or something. Don't know. Happy though.